### ISSUE TITLE
[BUG] Mixing access VLANs on a link should be prohibited

### ISSUE THREAD

**Issue by ipspace** [ipspace/netlab#1437]
Netlab accepts a lab topology that links two global VLANs on an access link. The results are "suboptimal". For example, the VLANs get different IP prefixes but all the VLAN interfaces get IP addresses from one of the VLANs.

IIRC this behavior arose from the idea that one could have VLAN domains limited to a single node, and that in that case it might make sense to link two such domains.

I think we should disable this capability as it only leads to confusion, or at the very minimum enforce that one cannot do this with two global VLANs while generating a warning saying "this is not supported, you are on your own" if someone does that with per-node VLANs.

This might be the root cause for #1436 and is probably related to #1432, so I'll wait for the feedback from @jbemmel before proceeding.

### Lab topology

```
---
module: [ vlan ]

defaults.device: eos

vlans:
  access_1:
  access_2:

nodes:
  a1:
  a2:

links:
- a1:
    vlan.access: access_1
  a2:
    vlan.access: access_2
```

**Comment by jbemmel** [comment]
The conflict arises because of ```mode: irb``` (default) and Netlab cannot pick IPs without violating its concept of a VLAN, right? With ```mode: bridge``` there would be no such inconsistency

I'd suggest to have Netlab generate an error in this case: "Cannot allocate IPs for cross-connected VLANS access_1 and access_2". Then we could allow the user to add a flag of some sort to ignore the error, such that the topology would be clearly marked as "unconventional" and we could avoid to waste time on troubleshooting it when submitted as bug or issue

**Comment by ipspace** [comment]
OK, I will:

* Generate a warning if **vlan.access** is defined on a node
* Generate a stricter warning if the **vlan.access** parameters defined on nodes do not match
* Add a warning to the VLAN documentation saying **vlan.access** parameter MUST be defined on a link, setting **vlan.access** on a node is unsupported and whatever behavior results from that **WILL NOT BE FIXED**.

**Comment by ipspace** [comment]
Slight change in plans:

* Warnings and errors are generated as needed
* Defining an access VLAN on a node interface is valid, as we might have a VLAN-aware node (router) connected to an access VLAN with a regular interface, so we have to define the access VLAN only on the switch, not on the link.

Will keep digging to fix #1436
